api/pkg/service/resource/resource.go

```diff
@@ -143,16 +138,24 @@ func (s *service) ByTypeName(ctx context.Context, p *resource.ByTypeNamePayload)
 	return s.resourcesForQuery(q)
 }
 
-func findOne(db *gorm.DB, result interface{}) error {
+// Find a resource using it's id
+func (s *service) ByID(ctx context.Context, p *resource.ByIDPayload) (res *resource.Resource, err error) {
 
-	err := db.Find(result).Error
-	if err != nil {
-		if gorm.IsRecordNotFoundError(err) {
-			return notFoundError
-		}
-		return fetchError
+	q := s.db.Scopes(withResourceDetails,
+		filterByID(p.ID))
+
+	var r model.Resource
+	if err := findOne(q, &r); err != nil {
+		return nil, err
 	}
-	return nil
+
+	res = initResource(r)
+
+	for _, v := range r.Versions {
+		res.Versions = append(res.Versions, minVersionInfo(v))
```

Comment:
`minVersionInfo` func has 4 fields but in this api's view we need only 2 so could you write a new func? `tinyVersionInfo` may be ?